pkg/vz/network_darwin.go

```diff
@@ -50,77 +51,117 @@ func DialQemu(unixSock string) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
+	vzConn := &packetConn{Conn: dgramConn}
 
-	remote := tcpproxy.DialProxy{
-		DialContext: func(context.Context, string, string) (net.Conn, error) {
-			return dgramConn, nil
-		},
-	}
-	go remote.HandleConn(qemuConn)
+	go forwardPackets(qemuConn, vzConn)
 
 	return client, nil
 }
 
-// QEMUPacketConn converts raw network packet to a QEMU supported network packet.
-type QEMUPacketConn struct {
-	unixConn net.Conn
+func forwardPackets(qemuConn *qemuPacketConn, vzConn *packetConn) {
```

Comment:
Do we need our custom forwardPackets ??

I would still prefer to use inetproxy itself. Unless it doesn't work even after wrapping 

We could simply wrap the packetConn with the fileconn. This way retry on ENOBUFS are present.
https://github.com/lima-vm/lima/blob/master/pkg/vz/network_darwin.go#L49

Reply:
> Do we need our custom forwardPackets ??

Yes, for 2 reasons:
- tcpproxy hides errors during io.Copy(), making debugging impossible
- we don't need tcpproxy since copying bytes between the sockets it trivial

> I would still prefer to use inetproxy itself. Unless it doesn't work even after wrapping

Can you explain why?

Looking at https://github.com/inetaf/tcpproxy:

> Proxy TCP connections based on static rules, HTTP Host headers, and SNI server names (Go package or binary)

We don't do any of that. We used a tiny bit of tcproxy copying bytes around, and this is better done in lima itself, where we can implement it in the best way for lima, and change it easily when needed.

> We could simply wrap the packetConn with the fileconn. This way retry on ENOBUFS are present. https://github.com/lima-vm/lima/blob/master/pkg/vz/network_darwin.go#L49

Adding another layer of wrapping to keep unneeded dependency does not sounds like the right way to me.